docs/tutorials/integrate-dotwallet.md

```diff
@@ -0,0 +1,143 @@
+```
+
+Here is a code example to create `DotwalletSigner` in `React`:
+
+
+```ts
+const provider = new DefaultProvider();
+
+// state is a random string to request login to DotWallet
+const state = localStorage.getItem("state")
+? (localStorage. getItem("state") as string)
+: crypto.randomUUID();
+
+localStorage.setItem("state", state);
+
+const accessToken = localStorage.getItem("access_token");
+
+const options = accessToken ? { accessToken } : {
+     client_id: "c152e571fffb163bc99bb87c51577354",
+     client_secret: "243720bb8bb573b3f07f6ba2838f4478",
+     redirect_uri: "http://localhost:3000/tic-tac-toe/",
+     state: state,
+};
+
+const signerRef = useRef(new DotwalletSigner(options, provider));
+
+const signer = signerRef.current;
+```
+
+
+Afterwards, you can use it as a signer like any other wallet as [before](../how-to-integrate-a-frontend.md#integrate-wallet). When the user clicks the login wallet button, the `requestAuth` method is called to access the wallet, and user is redirected to DotWallet's authorization page.
+
+
+```ts
+const walletLogin = async () => {
+     try {
+       const { isAuthenticated, error } = await signer. requestAuth();
+       if (!isAuthenticated) {
+         throw new Error(error);
+       }
+
+       ...
+
+       setConnected(true);
+     } catch (error) {
+       console.error("sensiletLogin failed", error);
+     }
+};
+```
+
+![](../../static/img/dotwallet-auth.png)
+
+After the user clicks `Agrees to authorize` to log in, the page will be redirected to the `redirect_uri` given by the app. In this example, `redirect_uri` is the current page <sup>2</sup>.
+We can tell whether it is a redirected page by seeing if the current query parameter has the `state` random string filled in when requesting to log in to the wallet. If yes, it will automatically call `requestAuth` to get `accessToken`. After successful authentication, `DotwalletSigner` will automatically save the obtained `accessToken` in `localStorage`. An authorized `DotwalletSigner` can then be created using the `DotwalletProductionOption`.
+
+```ts
+useEffect((login = walletLogin) => {
+     const urlParams = new URLSearchParams(window.location.search);
+     const state = urlParams. get("state");
+     // Check if redirected page
+     if (state && state === localStorage. getItem("state")) {
+         // it's a redirected page and create DotwalletSigner to get accessToken
+         login();
+     } else if (accessToken) {
+         //authorized
+         // create DotwalletSigner with accessToken
+         login();
+     }
+}, []);
+```
+
+After creating `DotwalletSigner` with `accessToken`, you can call all interfaces of `DotwalletSigner` like calling any other [`signer`](../how-to-test-a-contract.md#signer).
+
+
+Congrats! You have completed the integration of dotWallet.
+------
+
+[1] `client_secret` is usually stored in the backend service. Use the request `accessToken`.
```

Comment:
This is not clear. I do not understand. If `client_secret` is leaked, an attacker can use it to get an `accessToken`?